tests/tests/tier0/proxy-service-propagate-target-service-failure/test_proxy_service_propagate_target_service_failure.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import logging
+
+from typing import Dict
+from bluechi_test.test import BluechiTest
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+from bluechi_test.service import Option, Section, SimpleRemainingService
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+NODE_BAR = "node-bar"
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    node_foo = nodes[NODE_FOO]
+    node_bar = nodes[NODE_BAR]
+
+    service = SimpleRemainingService("target.service")
+    service.set_option(Section.Service, Option.ExecStart, "/bin/false")
+
+    node_bar.install_systemd_service(service)
+
+    service = SimpleRemainingService("requesting.service")
+    service.set_option(Section.Unit, Option.BindsTo, "bluechi-proxy@node-bar_target.service")
+    service.set_option(Section.Unit, Option.After, "bluechi-proxy@node-bar_target.service")
+    service.set_option(Section.Service, Option.ExecStart, "/bin/true")
+    service.set_option(Section.Service, Option.RemainAfterExit, "yes")
+
+    node_foo.install_systemd_service(service)
+
+    assert node_foo.systemctl.start_unit("requesting.service")
+    assert node_foo.wait_for_unit_state_to_be("requesting.service", "failed")
```

Comment:
```suggestion
    assert node_foo.wait_for_unit_state_to_be("requesting.service", "inactive")
```

Based on [BindsTo=](https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#BindsTo=):
```
if the unit bound to is stopped, this unit will be stopped too. 
```
This means, that `requesting.service` will be in state `inactive` (stopped) when the requested service `target.service` is stopped (by failure). 

You could add checks to
- `bluechi-proxy@node-bar_target.service` on node foo (should be `failed`, I think)
- and `bluechi-dep@target.service` on node bar (should be `inactive`, I think)